openmc/cmfd.py

```diff
@@ -0,0 +1,3072 @@
+        check_length('CMFD mesh upper_right', upper_right, 2, 3)
+        self._upper_right = upper_right
+
+    @dimension.setter
+    def dimension(self, dimension):
+        check_type('CMFD mesh dimension', dimension, Iterable, Integral)
+        check_length('CMFD mesh dimension', dimension, 2, 3)
+        for d in dimension:
+            check_greater_than('CMFD mesh dimension', d, 0)
+        self._dimension = dimension
+
+    @width.setter
+    def width(self, width):
+        check_type('CMFD mesh width', width, Iterable, Real)
+        check_length('CMFD mesh width', width, 2, 3)
+        for w in width:
+            check_greater_than('CMFD mesh width', w, 0)
+        self._width = width
+
+    @energy.setter
+    def energy(self, energy):
+        check_type('CMFD mesh energy', energy, Iterable, Real)
+        for e in energy:
+            check_greater_than('CMFD mesh energy', e, 0, True)
+        self._energy = energy
+
+    @albedo.setter
+    def albedo(self, albedo):
+        check_type('CMFD mesh albedo', albedo, Iterable, Real)
+        check_length('CMFD mesh albedo', albedo, 6)
+        for a in albedo:
+            check_greater_than('CMFD mesh albedo', a, 0, True)
+            check_less_than('CMFD mesh albedo', a, 1, True)
+        self._albedo = albedo
+
+    @map.setter
+    def map(self, meshmap):
+        check_type('CMFD mesh map', meshmap, Iterable, Integral)
+        for m in meshmap:
+            check_value('CMFD mesh map', m, [0, 1])
+        self._map = meshmap
+
+
+class CMFDRun(object):
+    r"""Class for running CMFD acceleration through the C API.
+
+    Attributes
+    ----------
+    tally_begin : int
+        Batch number at which CMFD tallies should begin accummulating
+    feedback_begin: int
+        Batch number at which CMFD feedback should be turned on
+    ref_d : list of floats
+        List of reference diffusion coefficients to fix CMFD parameters to
+    dhat_reset : bool
+        Indicate whether :math:`\widehat{D}` nonlinear CMFD parameters should
+        be reset to zero before solving CMFD eigenproblem.
+    display : dict
+        Dictionary indicating which CMFD results to output. Note that CMFD
+        k-effective will always be outputted. Acceptable keys are:
+
+        * "balance" - Whether to output RMS [%] of the resdiual from the
+          neutron balance equation on CMFD tallies (bool)
+        * "dominance" - Whether to output the estimated dominance ratio from
+          the CMFD iterations (bool)
+        * "entropy" - Whether to output the *entropy* of the CMFD predicted
+          fission source (bool)
+        * "source" - Whether to ouput the RMS [%] between the OpenMC fission
+          source and CMFD fission source (bool)
+
+    downscatter : bool
+        Indicate whether an effective downscatter cross section should be used
+        when using 2-group CMFD.
+    feedback : bool
+        Indicate or not the CMFD diffusion result is used to adjust the weight
```

Comment:
Nit Spelling: Did you mean "Indicate **whether** or not" rather than "Indicate or not"?